code-challenges.test.js

```diff
@@ -13,47 +22,121 @@
+        expect(randomNouns1.upperCase("Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew")
+        expect(randomNouns2.upperCase("Temperature", "Database", "Chopsticks", "Mango", "Deduction")
 
-// b) Create the function that makes the test pass.
+    })
+    // 
+    // Test had a faulty failure due to the return in line 103 which is blank. Not sure what is in that line. 
 
+    // b) Create the function that makes the test pass.
 
 
-// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+    const randomNouns(array) = => {
+        let randomNouns1 = ("streetlamp", "potato", "teeth", "conclusion", "nepheww");
 
-// a) Create a test with expect statements for each of the variables provided.
+        console.log("randomNouns1.toUpperCase())
 
-var vowelTester1 = "learn"
-// Expected output: 1
-var vowelTester2 = "academy"
-// Expected output: 0
-var vowelTester3 = "challenges"
-// Expected output: 2
+        temperature", "database", "chopsticks", "mango", "deduction
+        let randomNouns2 = (temperature", "database", "chopsticks", "mango", "deduction)
+        console.log(randomNouns1.toUpperCase);
 
 
+        // I got a litte confused on how to create this function. I save this one for last, my terminal locked up on VS 
 
-// b) Create the function that makes the test pass.
+
+        // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+
+        // create a function called firstVowel
+        // have function take in a parameter with string getConsoleOutput
+        // use if conditionalExpression
+        // use of .push and .index 
+
+        // a) Create a test with expect statements for each of the variables provided.
+
+
+        describe("firstVowel", () => {
+            var vowelTester1 = "learn"
+            expect(firtvowelTester1).toEqual(1)
+            expect(firtvowelTester1).toEqual(0)
+            expect(firtvowelTester1).toEqual(2)
+
+            var vowelTester2 = "academy"
+            // Expected output: 0
+            var vowelTester3 = "challenges"
+            // Expected output: 2
+
+            // b) Create the function that makes the test pass. 
+            const first vowel = string => {
+                let vowArr = []
+                if (string.indexOf("a") !== -1) {
+                    vowArr.push(string.indexOf("a"))
+                } if (string.indexOf("e") !== -1) {
+                    vowArr.push(string.indexOf("e"))
+                } if (string.indexOf("i") !== -1) {
+                    vowArr.push(string.indexOf("i"))
+                } if (string.indexOf("o") !== -1) {
+                    vowArr.push(string.indexOf("o"))
+                } if (string.indexOf("u") !== -1) {
+                    vowArr.push(string.indexOf("u"))
+                }
+                return Math.min(...vowArr)
+            }
```

Comment:
Your logic looks good, but it looks like your function is inside the test. Make sure you are checking your code super often!